### PR TITLE
feat(client-debugger): Differentiate IDebuggerMessage and ISourcedDebuggerMessage

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

Separates `IDebuggerMessage` into `IDebuggerMessage` (without the `source` property) and `ISourcedDebuggerMessage` (with it).

This is prep work split from #14628 to make that one easier to review. The changes there leverage the move of the `source` property so message senders don't need to provide it when creating their messages, as it gets stamped automatically by the "pipes" that take care of message passing.

This was mostly a rename-all, with a few adjustments needed in exports.

### CODE REVIEW

**Review by Josmithr — APPROVED** [review]
Should the various message interfaces be standard `IDebuggerMessages`, rather than sourced ones? Otherwise we won't be able to send them unsourced to the message relay, etc. Or will that change come later?

**Comment by alexvy86** [comment]
> Should the various message interfaces be standard `IDebuggerMessages`, rather than sourced ones? Otherwise we won't be able to send them unsourced to the message relay, etc. Or will that change come later?

Yeah, I was planning to make that a separate step so this was also easy to review as basically just a rename. I'll hold on merging this until we merge #14507, so I can apply the change to the new messages from that one.

**Comment by alexvy86** [comment]
Now that Ji's and Tong's PR were merged into main, I updated this PR to make the necessary adjustments on the new messages from those two PRs. I'll merge it once CI passes.

**Pull request merged**
